content/200-concepts/200-database-connectors/07-mongodb.mdx

```diff
@@ -188,3 +188,36 @@ To resolve this, we suggest you change your deployment to one with a replica set
 One simple way for this is to use [MongoDB Atlas](https://www.mongodb.com/cloud/atlas) to launch a free instance that has replica set support out of the box.
 
 There's also an option to run the replica set locally with this guide: https://docs.mongodb.com/manual/tutorial/convert-standalone-to-replica-set
+
+## Type mapping between MongoDB and the Prisma schema
+
+> Alternatively, see the [Prisma schema reference](/reference/api-reference/prisma-schema-reference#model-field-scalar-types) for type mappings organized by Prisma type.
+
+The MongoDB connector maps the [scalar types](/concepts/components/prisma-schema/data-model#scalar-fields) from the Prisma [data model](/concepts/components/prisma-schema/data-model) to MongoDB's native column types as follows:
+
+| Prisma    | MongoDB      |
+| --------- | ------------ |
+| `String`  | `String`     |
+| `Bool`    | `Boolean`    |
+| `Double`  | `Double`     |
+| `Int32`   | `Int32`      |
+| `Int64`   | `Int64`      |
+| `Decimal` | `Decimal128` |
+| `Date`    | `DateTime`   |
+| `BinData` | `Binary`     |
```

Comment:
Need to add Json to this list - how do I interpret the mapping [here](https://github.com/prisma/prisma-engines/blob/ccf3dc944acdabb431947150e12b984b34c538cd/introspection-engine/connectors/mongodb-introspection-connector/src/sampler/field_type.rs#L37)?

Also, is the [mapping in the schema reference](http://localhost:8000/reference/api-reference/prisma-schema-reference#mongodb-preview-7) still correct?

Reply:
- You don't interpret this mapping at all in my opinion, this is internal conversion code and uses a non-Prisma schema type typing. Prisma schema doesn't have `Int32` or `Int64`, for example.

- Second link goes to localhost.

Reply:
Going by [this reference](https://www.mongodb.com/docs/v5.0/reference/bson-types/) as a base.

In my opinion, the mapping should be in reverse and be the following (the table denotes _valid_ mappings):
| MongoDB    | Prisma      |
| --------- | ------------ |
| `String`  | `String`     |
| `ObjectId`  | `String` or `Binary`     |
| `Bool`    | `Boolean`    |
| `Double`  | `Double`     |
| `Int`   | `Int`      |
| `Long`   | `Int` or `BigInt`      |
| `Date`    | `DateTime`   |
| `Timestamp`    | `DateTime`   |
| `BinData` | `Binary`     |

Unsupported MongoDB types (e.g. they do not have a valid Prisma mapping as of now):
- `Decimal128`
- `Undefined`
- `DBPointer`
- `Null`
- `Symbol`
- `MinKey`
- `MaxKey`
- `Object`
- `Javascript`
- `JavascriptWithScope`
- `Regex`

Note:
- There's also `Array` on the Mongo side, which get's translated as Prisma list type of the whatever the inner type is (e.g. `String[]`).
- Prismas `Json` type can be used to emulate an `Any` type, but will not always work. Some types and values can not be converted into JSON, see the [Extended JSON reference](https://www.mongodb.com/docs/v5.0/reference/mongodb-extended-json/).

Reply:
Other connectors currently have the table under a "Migrations" headline: https://www.prisma.io/docs/concepts/database-connectors/mysql#type-mapping-between-mysql-to-prisma-schema So there it means "If I have this in my Prisma schema, what will it be translated to in my database when I migrate?"

Reply:
Ah, ok, thanks. I found the 'Migrations' headline very confusing and had no idea what it referred to (as it was just that single word). Have taken it out for now and will either replace it with a different heading or include some words of explanation of what it means.

The heading to the second table on the other connectors, 'Native type mappings', is also confusing - native to Prisma or native to the underlying database?

Reply:
An annoying thing about having the mapping order as Prisma -> MongoDB is that there's no easy to way to include MongoDB types which are unsupported in Prisma

Would these go in a separate list, if we kept the ordering the same as for the other connectors?

Reply:
I've made some updates:

- added table headings back in but made them more descriptive
- kept the mapping the current way round, and copied the list of Prisma types from the Postgres connector
- added a list of Mongo types unsupported in Prisma

Reply:
I think those tables (and all other places that describe scalar types or native types are definitely up for a complete rework, see my PR where I suggested extracting some of it into its own page: https://github.com/prisma/docs/pull/2887 I think it would be a _great_ idea to do a project soon to fix that some way and would be happy to help9